run/jenkins.py

```diff
@@ -0,0 +1,282 @@
+    )
+
+    if platform.get('sauce'):
+        if args['sauce_from_metadata']:
+            print 'Fetching Sauce creds from GCE metadata...'
+
+            for key in ('sauce_user', 'sauce_key'):
+                url = '%s/project/attributes/%s' % (METADATA_URL, key)
+                res = requests.get(url, headers={'Metadata-Flavor': 'Google'})
+                args[key] = res.text
+                assert args[key], 'Metadata key %s is empty.' % key
+
+        assert args['sauce_key'], 'SAUCE_KEY env var required'
+        assert args['sauce_user'], 'SAUCE_USER env var required'
+        SAUCE_TUNNEL_ID = '%s_%s' % (platform_id, int(time.time()))
+
+    assert len(args['SHA']) == 10, 'SHA must a WPT SHA[:10]'
+
+    # Hack because Sauce expects a different name
+    # Maybe just change it in browsers.json?
+    if platform['browser_name'] == 'edge':
+        sauce_browser_name = 'MicrosoftEdge'
+    else:
+        sauce_browser_name = platform['browser_name']
+    product = 'sauce:%s:%s' % (sauce_browser_name, platform['browser_version'])
+
+    patch_wpt(args['wptd_path'], args['wpt_path'], platform)
+
+    if platform.get('sauce'):
+        command = [
+            './wpt', 'run', product,
+            '--sauce-platform=%s' % platform['os_name'],
+            '--sauce-key=%s' % args['sauce_key'],
+            '--sauce-user=%s' % args['sauce_user'],
+            '--sauce-tunnel-id=%s' % SAUCE_TUNNEL_ID,
+            '--no-restart-on-unexpected',
+            '--processes=2',
+            '--run-by-dir=3',
+            '--log-mach=-',
+            '--log-wptreport=%s' % LOCAL_REPORT_FILEPATH,
+            '--install-fonts'
+        ]
+        if args['run_path']:
+            command.insert(3, args['run_path'])
+    else:
+        command = [
+            'xvfb-run', '--auto-servernum',
+            './wpt', 'run',
+            platform['browser_name'],
+            '--install-fonts',
+            '--install-browser',
+            '--yes',
+            '--log-mach=%s' % LOCAL_LOG_FILEPATH,
+            '--log-wptreport=%s' % LOCAL_REPORT_FILEPATH,
+        ]
+        if args['run_path']:
+            command.insert(5, args['run_path'])
+
+    if args['jenkins_job_name'] and '_of_' in args['jenkins_job_name']:
+        # Expects a job name like chunk_2_of_4
+        chunks = re.findall(r'(\d+)_of_(\d+)', args['jenkins_job_name'])
+        chunk_num, total_chunks = chunks[0]
+        command.extend([
+            '--this-chunk=%s' % chunk_num,
+            '--total-chunks=%s' % total_chunks,
+        ])
+
+    return_code = subprocess.call(command, cwd=args['wpt_path'])
+
+    print('==================================================')
+    print('Finished WPT run')
+    print('Return code from wptrunner: %s' % return_code)
+
+    with open(LOCAL_REPORT_FILEPATH) as f:
+        report = json.load(f)
+
+    assert len(report['results']) > 0, (
+        '0 test results, something went wrong, stopping.')
```

Comment:
Any hints at where to look for logged errors to diagnose?

Reply:
Where the actual logs are? You can find them at URLs like https://ci.wpt.fyi/job/edge-latest-stable/17/console

Reply:
Perhaps the logs go to the same place as this error message. I assumed that the WPT run occurs in some different context and that its output might not be in the same place this message shows up. Is that false? If so, please disregard.

Reply:
Yep, they go to the same place. We funnel the stdout of the WPT run to the stdout of `run/jenkins.py` by passing `--log-mach=-` to `wpt run`.